python/speech-to-text/main.py

```diff
@@ -0,0 +1,83 @@
+import os
+import json
+import wave
+import base64
+from googleMain import google_speech_recognition
+from amazonMain import amazon_recognize
+from azureSpeech import azure_recognize
+
+def write_wav(audioBase64, fileName):
+    base_path = os.path.dirname(os.path.abspath(__file__))
+    filePath = os.path.join(base_path, "tempAudio",fileName)
```

Comment:
If this function is executed concurrently, there can be conflicts with this path. The best thing to do is to make sure you're writing to a temp path using [tempfile](https://docs.python.org/3/library/tempfile.html).